File: codegen/results/v002/gemma4_31b_think/20260414_091539/go/004_json_filter/solution.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"sort"
)

// Record defines the structure of the input JSON objects
type Record struct {
	Name   string  `json:"name"`
	Age    int     `json:"age"`
	Active bool    `json:"active"`
	Score  float64 `json:"score"`
}

func main() {
	// Read the input file
	data, err := os.ReadFile("input/data.json")
	if err != nil {
		os.Exit(1)
	}

	// Parse JSON array
	var records []Record
	if err := json.Unmarshal(data, &records); err != nil {
		os.Exit(1)
	}

	// Filter records: active is true and age is 30 or older
	var filtered []Record
	for _, r := range records {
		if r.Active && r.Age >= 30 {
			filtered = append(filtered, r)
		}
	}

	// Sort by name ascending
	sort.Slice(filtered, func(i, j int) bool {
		return filtered[i].Name < filtered[j].Name
	})

	// Marshal result to JSON
	output, err := json.Marshal(filtered)
	if err != nil {
		os.Exit(1)
	}

	// Output to stdout
	fmt.Println(string(output))
}

File: codegen/results/v002/gemma4_31b_think/20260414_091539/typescript/004_json_filter/solution.ts
import * as fs from 'fs';
import * as path from 'path';

interface Record {
  name: string;
  age: number;
  active: boolean;
  score: number;
}

function solve() {
  try {
    // Define path to input file
    const inputPath = path.join(process.cwd(), 'input', 'data.json');
    
    // Read the file content
    const rawData = fs.readFileSync(inputPath, 'utf8');
    
    // Parse JSON
    const data: Record[] = JSON.parse(rawData);
    
    // Filter: active is true AND age is 30 or older
    const filtered = data.filter(item => item.active === true && item.age >= 30);
    
    // Sort: name ascending
    const sorted = filtered.sort((a, b) => {
      if (a.name < b.name) return -1;
      if (a.name > b.name) return 1;
      return 0;
    });
    
    // Output JSON array to stdout
    process.stdout.write(JSON.stringify(sorted, null, 2) + '\n');
  } catch (error) {
    // In a production environment, we would handle specific errors.
    // The requirement says "Do not output anything else", so we stay silent 
    // or let the runtime handle the exception.
    process.exit(1);
  }
}

solve();